### PR TITLE
feat: add custom status readers

### PR DESCRIPTION
**Pull request by zreigz**
Get status directly from the Deployment, ReplicaSet and StatefulSet objects

### CODE REVIEW

File: internal/kstatus/statusreaders/common.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+func (b *baseStatusReader) Supports(gk schema.GroupKind) bool {
+	return b.resourceStatusReader.Supports(gk)
+}
+
+// ReadStatus reads the object identified by the passed-in identifier and computes it's status. It reads
+// the resource here, but computing status is delegated to the ReadStatusForObject function.
+func (b *baseStatusReader) ReadStatus(ctx context.Context, reader engine.ClusterReader, identifier object.ObjMetadata) (*event.ResourceStatus, error) {
+	object, err := b.lookupResource(ctx, reader, identifier)
```
Comment (michaeljguarino) [inline internal/kstatus/statusreaders/common.go:51]: why is this lookup needed?  If it's needed that's ultimately fine, but seems like the unstructured resource should have been available and could have been avoided.

File: internal/kstatus/statusreaders/deployment.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+func (d *deploymentResourceReader) Supports(gk schema.GroupKind) bool {
+	return gk == appsv1.SchemeGroupVersion.WithKind("Deployment").GroupKind()
+}
+
+func (d *deploymentResourceReader) ReadStatusForObject(_ context.Context, _ engine.ClusterReader,
+	deployment *unstructured.Unstructured) (*event.ResourceStatus, error) {
+	identifier := object.UnstructuredToObjMetadata(deployment)
+
+	res, err := status.Compute(deployment)
```
Comment (michaeljguarino) [inline internal/kstatus/statusreaders/deployment.go:42]: isn't the code that lists the pods w/in kstatus?  I would assume this just means we'll execute it by delegating to it.

File: internal/kstatus/statusreaders/common.go (hunk 1 above)
Comment (zreigz) [reply internal/kstatus/statusreaders/common.go:51]: It resolves the GroupKindVersion for the given identifier and check if the resource exists

File: internal/kstatus/statusreaders/deployment.go (hunk 2 above)
Comment (zreigz) [reply internal/kstatus/statusreaders/deployment.go:42]: This code looks at the properties that are available on all or most of the Kubernetes resources. I got rid of the code for checking the Pod's status. It checks only the main resource type

**Review by michaeljguarino — APPROVED** [review]

**Pull request merged**